src/HDF5.jl

```diff
@@ -778,6 +791,7 @@ checkvalid(obj) = isvalid(obj) ? obj : error("File or object has been closed")
 # Close functions that should try calling close regardless
 function close(obj::HDF5File)
     if obj.id != -1
+        flush(obj)
```

Comment:
is this something we forgot ?

Reply:
This is only necessary when `fclose_degree` is `H5F_CLOSE_WEAK`. I think libhdf5 will handle this internally when `fclose_degree` is `H5F_CLOSE_STRONG`.